web-console/knikubevirt/vm-list/vm-list.md

```diff
@@ -0,0 +1,85 @@
+
+The VM list contains columns for the following:
+- Name
+- Namespace
+- Status
+  - Running, Off, Error (with custom messages), Cloning, Migrating, Importing, Warning, Pending (with changes that will apply after restart), Unknown
+- Created
+  - Date formatting should follow OpenShift's console, viewable [here](https://github.com/openshift/console/blob/master/frontend/public/components/utils/timestamp.jsx) and [here](https://github.com/openshift/console/blob/master/frontend/public/components/utils/datetime.ts)
+  - If possible, a tooltip should appear on hover with the exact creation date, including year.
+- Node
+- IP Address
+- FQDN
+
+## Status tooltips
+
+![VM list status tooltips](img/0-0a-tooltips.png)
+
+Statuses other than Running or Off should be colored blue to indicate that they can be hovered and/or clicked.
+
+Hovering over a blue status should display a tooltip with a brief description of the issue or underlying process. If multiple errors/warnings exist, the description should be more generic. Clicking on a blue status should bring the user to the VM's Events page for more details.
+
+## Filtering
+
+### Item filters
+
+![filters](img/1-0-filtering.png)
+
+The predefined filters in the filter bar can be used to narrow the VM list by most statuses. The user clicks each filter to toggle it on and off.
+
+### Name filter
+
+![Name filter](img/1-1-filtering-name.png)
+
+The name filter can also be used to further narrow down the VM list. Name filters do not override the filters set in the filtering bar, meaning hidden items will remain hidden even if searched for.
+
+(Note: This behavior could be a usability issue if users expect the title filter input to behave like a traditional search input.)
+
+### Search filter (concept)
+
+![Searching by IP](img/1-2-filtering-search-ip.png)
+
+If possible, the name filter could become a traditional search filter instead, allowing the user to filter results by any text string in any column. This could enable users to quickly find a VM by IP address, FQDN, or possibly other fields in the future. Multiple filters could be delineated by a comma in the search query. A help icon describing this advanced functionality would likely be necessary.
+
+![Searching by status](img/1-3-filtering-search-status.png)
+
+## Sorting
+
+VM lists are sorted alphabetically by name by default. Clicking on a different column title will sort that column the same way. Clicking on the same column title twice will reverse the sort direction.
+
+![Sort by name](img/2-0-sort-name.png)
+
+![Sort by age](img/2-1-sort-node.png)
+
+![Sort by reverse age](img/2-2-sort-node-reverse.png)
+
+## States and Actions
+
+When a VM is off, the actions kebab contains options to Run, Edit, Migrate, Clone, Create Template, Take Snapshot, and Delete the VM.
+
+![VM actions while off](img/3-0-selected-vm-off.png)
+
+When a VM is running the Run option is replaced with two dropdowns for Power and Connect to Console.
```

Comment:
We don't do anything like this today but I can see the need for sure. We may want to run by Sam. Curious what other OpenShift designers think!